nf_core/modules/lint/main_nf.py

```diff
@@ -220,14 +220,18 @@ def check_process_section(self, lines):
             lspl = l.lstrip("https://").split(":")
             if len(lspl) == 2:
                 # e.g. 'https://containers.biocontainers.pro/s3/SingImgsRepo/biocontainers/v1.2.0_cv1/biocontainers_v1.2.0_cv1.img' :
-                singularity_tag = "_".join(lspl[0].split("/")[-1].strip().rstrip(".img").split("_")[1:])
+                if "YOUR-TOOL-HERE" not in lspl[0]:
+                    # Get singularity container version when a container is provided
+                    singularity_tag = re.search("biocontainers_(.*).img", lspl[0]).group(1)
             else:
                 # e.g. 'https://depot.galaxyproject.org/singularity/fastqc:0.11.9--0' :
                 singularity_tag = lspl[-2].strip()
         if l.startswith("biocontainers/") or l.startswith("quay.io/"):
             # e.g. 'quay.io/biocontainers/krona:2.7.1--pl526_5' }"
             # e.g. 'biocontainers/biocontainers:v1.2.0_cv1' }"
-            docker_tag = l.split(":")[-1].strip("}").strip()
+            if "YOUR-TOOL-HERE" not in l:
+                # Get docker container version when a container is provided
+                docker_tag = re.search("\:(.*) \}", l).group(1)
```

Comment:
I think this will pick up the trailing single quote, which we don't want..? (though maybe the original code also did that?)

This should be a little less strict on whitespace and get only the tag without the quote mark:
```suggestion
                docker_tag = re.search("\:([^\s\'\"]+)[\'\"]\s*\}", l).group(1)
```

Reply:
Single and double quotes were removed before (lines 215 & 216), but taking some inspiration from your suggestions, I think it may be better to maintain them and use those modifications in the regex, I mix-and-matched between your different suggestions, hope it's okay :)